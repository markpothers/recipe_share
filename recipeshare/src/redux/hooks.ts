import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux"
import type { AppDispatch } from "./store"
import type { RootStateType } from "./types"

export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootStateType> = useSelector
