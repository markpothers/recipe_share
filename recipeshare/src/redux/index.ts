export { useAppDispatch, useAppSelector } from "./hooks";
export * from "./types";
export type { RootState } from "./store"
export {
	updateSingleRecipeList,
	updateAllRecipeLists,
	updateSingleChefList,
	updateAllChefLists,
	storeRecipeDetails,
	updateNewUserDetails,
	clearNewUserDetails,
	updateLoginUserDetails,
	clearLoginUserDetails,
	addRecipeLike,
	removeRecipeLike,
	addRecipeReShare,
	removeRecipeReShare,
	addRecipeMake,
	addMakePic,
	addMakePicChef,
	saveRemainingMakePics,
	updateComments,
	storeNewFollowers,
	updateLoggedInChef,
	storeChefDetails,
	clearChefDetails,
	stayLoggedIn
} from "./rootReducer"
export { default as rootReducer } from "./rootReducer"
