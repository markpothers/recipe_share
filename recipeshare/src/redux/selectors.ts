import { ReduxStateType } from "./types";
import { createSelector } from "@reduxjs/toolkit";

const selectRoot = (state: ReduxStateType) => state.root

export const getLoggedInChef = createSelector(selectRoot, root => root.loggedInChef)
export const getLoggedInUserDetails = createSelector(selectRoot, root => root.loginUserDetails)
export const getDeviceType = createSelector(selectRoot, root => root.deviceType)
export const getStayLoggedIn = createSelector(selectRoot, root => root.stayLoggedIn)
