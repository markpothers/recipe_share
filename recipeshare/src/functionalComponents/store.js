import { createStore, compose, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk'
import reducer from './reducer'

const initialState = {
  loggedInChef: {
    id: "",
    username: "",
    auth_token: "",
    imageURL: "",
    is_admin: false
  },
  recipes: {
    all: [],
    chef: [],
    chef_feed: [],
    chef_liked: [],
    chef_made: [],
    global_ranks: [],
    most_liked: [],
    most_made: []
  },
  recipes_details: {

  },
  newRecipeDetails: {
    name: "",
    instructions: ['test', 'test2'],
    ingredients: {
      ingredient1 :{
        name:"",
        quantity: "",
        unit: "Oz"
      }
    },
    difficulty: "0",
    time: "00:15",
    imageBase64: "",
    filter_settings: {
      "Breakfast": false,
      "Lunch": false,
      "Dinner": false,
      "Chicken": false,
      "Red meat": false,
      "Seafood": false,
      "Vegetarian": false,
      "Salad": false,
      "Vegan": false,
      "Soup": false,
      "Dessert": false,
      "Side": false,
      "Whole 30": false,
      "Paleo": false,
      "Freezer meal": false,
      "Keto": false,
      "Weeknight": false,
      "Weekend": false,
      "Gluten free": false,
      "Bread": false,
      "Dairy free": false,
      "White meat": false,
    },
    cuisine: "Any",
    serves: "Any",
    acknowledgement: ""
  },
  newUserDetails: {
    first_name: "",
    last_name: "",
    username: "",
    e_mail: "",
    password: "",
    password_confirmation: "",
    country: "United States",
    imageURL: "",
    profile_text: ""
  },
  loginUserDetails: {
    e_mail: "markpothers@hotmail.com",
    password: "123456"
  },
  chefs: {
    all_chefs: [],
    followed: [],
    most_liked_chefs: [],
    most_made_chefs: [],
    chef_followees: [],
    chef_followers: []
  },
  chefs_details: {},
  filter_settings: {
    "Breakfast": false,
    "Lunch": false,
    "Dinner": false,
    "Chicken": false,
    "Red meat": false,
    "Seafood": false,
    "Vegetarian": false,
    "Salad": false,
    "Vegan": false,
    "Soup": false,
    "Dessert": false,
    "Side": false,
    "Whole 30": false,
    "Paleo": false,
    "Freezer meal": false,
    "Keto": false,
    "Weeknight": false,
    "Weekend": false,
    "Gluten free": false,
    "Bread": false,
    "Dairy free": false,
    "White meat": false,
  },
  cuisine: "Any",
  serves: "Any",
  stayLoggedIn: false,
}

  const middleware = compose(
    applyMiddleware(ReduxThunk)
  )

  export const store = createStore(reducer, initialState, middleware)
