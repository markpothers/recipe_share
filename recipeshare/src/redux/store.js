import { createStore, compose, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk'
import reducer from './reducer'

// import { filters } from '../dataComponents/filters'

//exported ONLY for unit testing
export const initialState = {
	loggedInChef: {
		id: "",
		username: "",
		auth_token: "",
		image_url: "",
		is_admin: false
	},
	allRecipeLists: {},
	allChefLists: {},
	recipes_details: {

	},
	newRecipeDetails: {
		name: "",
		instructions: [
			'',
			'',
			'',
			'',
		],
		instructionImages: [],
		ingredients: {
			ingredient1: {
				name: "",
				quantity: "",
				unit: "Oz"
			}
		},
		difficulty: "0",
		time: "00:15",
		imageBase64: "",
		filter_settings: {
			"Breakfast": false,
			"Lunch": false,
			"Dinner": false,
			"Chicken": false,
			"Red meat": false,
			"Seafood": false,
			"Vegetarian": false,
			"Salad": false,
			"Vegan": false,
			"Soup": false,
			"Dessert": false,
			"Side": false,
			"Whole 30": false,
			"Paleo": false,
			"Freezer meal": false,
			"Keto": false,
			"Weeknight": false,
			"Weekend": false,
			"Gluten free": false,
			"Bread": false,
			"Dairy free": false,
			"White meat": false,
		},
		cuisine: "Any",
		serves: "Any",
		acknowledgement: ""
	},
	newUserDetails: {
		first_name: "",
		last_name: "",
		username: "",
		e_mail: "",
		password: "",
		password_confirmation: "",
		country: "United States",
		image_url: "",
		profile_text: ""
	},
	loginUserDetails: {
		e_mail: "",
		password: ""
	},
	chefs: {
		all_chefs: [],
		followed: [],
		most_liked_chefs: [],
		most_made_chefs: [],
		chef_followees: [],
		chef_followers: []
	},
	chefs_details: {},
	filter_settings: {
		"Breakfast": false,
		"Lunch": false,
		"Dinner": false,
		"Chicken": false,
		"Red meat": false,
		"Seafood": false,
		"Vegetarian": false,
		"Salad": false,
		"Vegan": false,
		"Soup": false,
		"Dessert": false,
		"Side": false,
		"Whole 30": false,
		"Paleo": false,
		"Freezer meal": false,
		"Keto": false,
		"Weeknight": false,
		"Weekend": false,
		"Gluten free": false,
		"Bread": false,
		"Dairy free": false,
		"White meat": false,
	},
	filterCuisines: {
		all: "Any",
		chef: "Any",
		chef_feed: "Any",
		chef_liked: "Any",
		chef_made: "Any",
		global_ranks: "Any",
		most_liked: "Any",
		most_made: "Any"
	},
	serves: "Any",
	stayLoggedIn: false,
}

export const middleware = compose( //exported ONLY for unit testing
	applyMiddleware(ReduxThunk)
)

export const store = createStore(reducer, initialState, middleware)
