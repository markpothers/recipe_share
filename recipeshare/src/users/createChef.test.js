import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Provider } from 'react-redux'
import { store } from '../redux/store'

import CreateChef from './createChef.js'
import { TouchableOpacity, TextInput } from 'react-native';


describe('CreateChef', () => {

	let component
	let route

	beforeAll(() => {
		// console.log('runs at the beginning of everything')
	})

	beforeEach(() => {
		// console.log('runs before every test')

		route = {
			params: {
				successfulLogin: false
			}
		}
	})

	afterEach(() => {
		// console.log('runs after each test')
	})

	afterAll(() => {
		// console.log('runs after all tests have completed')
	})

	test('renders; has 5 TextInputs and 9 TouchableOpacities', () => {
		act(() => {
			component = renderer.create(
				<Provider
					store={store}
				>
					<CreateChef
						route={route}
					/>
				</Provider>
			)
		})
		const image = component.toJSON()
		expect(image).toMatchSnapshot()
		let root = component.root
		const inputs = root.findAllByType(TextInput)
		expect(inputs.length).toEqual(5)
		const buttons = root.findAllByType(TouchableOpacity)
		expect(buttons.length).toEqual(10)
	})

});
