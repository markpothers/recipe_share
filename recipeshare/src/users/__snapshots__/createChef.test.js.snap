// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateChef renders; has 5 TextInputs and 9 TouchableOpacities 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "borderColor": "blue",
      "borderWidth": 0,
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "red",
          "borderWidth": 0,
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 0,
          "height": 1334,
          "position": "absolute",
          "width": 750,
        }
      }
    />
    <RCTScrollView
      keyboardShouldPersistTaps="always"
      nestedScrollEnabled={true}
      scrollEnabled={true}
      style={
        Object {
          "borderColor": "yellow",
          "borderWidth": 0,
          "flex": 1,
        }
      }
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderWidth": 0,
                  "justifyContent": "center",
                  "marginLeft": 75,
                  "marginRight": 75,
                  "width": 600,
                },
                Object {
                  "marginTop": 66.7,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <Text
                  maxFontSizeMultiplier={2}
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "color": "#505050",
                      "fontSize": 33.35,
                      "fontWeight": "bold",
                      "paddingLeft": 15,
                      "paddingRight": 15,
                      "textAlign": "center",
                      "width": "100%",
                    }
                  }
                >
                  Please register and click the link in your confirmation e-mail!
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    keyboardType="email-address"
                    maxFontSizeMultiplier={2}
                    onChange={[Function]}
                    placeholder="e-mail"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "borderRadius": 5,
                        "fontSize": 30.595932917809698,
                        "paddingLeft": 15,
                        "paddingVertical": 6.67,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                        "width": "100%",
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    maxFontSizeMultiplier={2}
                    onChange={[Function]}
                    placeholder="username"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "borderRadius": 5,
                        "fontSize": 30.595932917809698,
                        "paddingLeft": 15,
                        "paddingVertical": 6.67,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                        "width": "100%",
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "white",
                        "borderColor": "#104e01",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "justifyContent": "center",
                        "overflow": "hidden",
                        "width": "100%",
                      },
                      Object {
                        "height": 44,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "height": "100%",
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "height": "100%",
                          "justifyContent": "flex-start",
                          "opacity": 1,
                          "width": "100%",
                        }
                      }
                    >
                      <Text
                        maxFontSizeMultiplier={2}
                        style={
                          Object {
                            "marginLeft": 15,
                          }
                        }
                      >
                        United States
                      </Text>
                      <Icon
                        name="caret-down"
                        size={46.690000000000005}
                        style={
                          Array [
                            Object {
                              "color": "#505050",
                            },
                            Object {
                              "marginRight": "5%",
                              "position": "absolute",
                              "right": 0,
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "white",
                        "borderColor": "#104e01",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "height": "100%",
                        "overflow": "hidden",
                        "width": "100%",
                      },
                      Object {
                        "minHeight": 160.07999999999998,
                      },
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    maxFontSizeMultiplier={2}
                    multiline={true}
                    numberOfLines={3}
                    onChange={[Function]}
                    placeholder="about me"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "borderRadius": 5,
                        "fontSize": 30.595932917809698,
                        "paddingLeft": 15,
                        "paddingVertical": 6.67,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                        "width": "100%",
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    maxFontSizeMultiplier={2}
                    onChange={[Function]}
                    placeholder="password"
                    rejectResponderTermination={true}
                    secureTextEntry={true}
                    style={
                      Object {
                        "borderRadius": 5,
                        "fontSize": 30.595932917809698,
                        "paddingLeft": 15,
                        "paddingVertical": 6.67,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                        "width": "100%",
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "overflow": "hidden",
                      "width": "100%",
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    maxFontSizeMultiplier={2}
                    onChange={[Function]}
                    placeholder="password confirmation"
                    rejectResponderTermination={true}
                    secureTextEntry={true}
                    style={
                      Object {
                        "borderRadius": 5,
                        "fontSize": 30.595932917809698,
                        "paddingLeft": 15,
                        "paddingVertical": 6.67,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                        "width": "100%",
                      }
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "minHeight": 80.03999999999999,
                      "opacity": 1,
                      "width": 360,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Object {
                        "borderRadius": 5,
                        "fontSize": 30.595932917809698,
                        "paddingLeft": 15,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                        "width": "100%",
                      }
                    }
                  >
                    View Terms & Conditions
                  </Text>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff59b",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "space-evenly",
                      "maxWidth": 225,
                      "minHeight": 80.03999999999999,
                      "minWidth": 225,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "flexWrap": "wrap",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      maxFontSizeMultiplier={2}
                      style={
                        Object {
                          "color": "#104e01",
                          "fontSize": 27.53633962602873,
                          "fontWeight": "bold",
                          "paddingHorizontal": 7.5,
                          "textAlign": "center",
                        }
                      }
                    >
                      I accept
                    </Text>
                    <RCTSwitch
                      accessibilityRole="switch"
                      onChange={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      onTintColor="#4b714299"
                      style={
                        Object {
                          "height": 31,
                          "width": 51,
                        }
                      }
                      thumbTintColor="#ececec"
                      value={false}
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "minHeight": 80.03999999999999,
                      "opacity": 1,
                      "width": 360,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Object {
                        "borderRadius": 5,
                        "fontSize": 30.595932917809698,
                        "paddingLeft": 15,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                        "width": "100%",
                      }
                    }
                  >
                    View Privacy Policy
                  </Text>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff59b",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "space-evenly",
                      "maxWidth": 225,
                      "minHeight": 80.03999999999999,
                      "minWidth": 225,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "flexWrap": "wrap",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      maxFontSizeMultiplier={2}
                      style={
                        Object {
                          "color": "#104e01",
                          "fontSize": 27.53633962602873,
                          "fontWeight": "bold",
                          "paddingHorizontal": 7.5,
                          "textAlign": "center",
                        }
                      }
                    >
                      I accept
                    </Text>
                    <RCTSwitch
                      accessibilityRole="switch"
                      onChange={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      onTintColor="#4b714299"
                      style={
                        Array [
                          Object {
                            "height": 31,
                            "width": 51,
                          },
                          Object {
                            "transform": Array [
                              Object {
                                "scaleX": 0.7,
                              },
                              Object {
                                "scaleY": 0.7,
                              },
                            ],
                          },
                        ]
                      }
                      thumbTintColor="#ececec"
                      value={false}
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff59b",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "center",
                      "maxWidth": "100%",
                      "minHeight": 80.03999999999999,
                      "minWidth": 285,
                      "opacity": 1,
                      "width": "100%",
                    }
                  }
                >
                  <Icon
                    name="camera"
                    size={25}
                    style={
                      Object {
                        "color": "#104e01",
                      }
                    }
                  />
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Array [
                        Object {
                          "color": "#104e01",
                          "fontSize": 27.53633962602873,
                          "fontWeight": "bold",
                          "paddingHorizontal": 7.5,
                          "textAlign": "center",
                        },
                        Object {
                          "marginLeft": 22.5,
                        },
                      ]
                    }
                  >
                    Add profile picture
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "borderColor": "yellow",
                  "borderWidth": 0,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                    "marginTop": 6.67,
                    "minHeight": 80.03999999999999,
                    "overflow": "hidden",
                    "width": "100%",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff59b",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "space-evenly",
                      "maxWidth": "50%",
                      "minHeight": 80.03999999999999,
                      "minWidth": 285,
                      "opacity": 1,
                    }
                  }
                >
                  <Icon
                    name="login"
                    size={25}
                    style={
                      Object {
                        "color": "#104e01",
                      }
                    }
                  />
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Object {
                        "color": "#104e01",
                        "fontSize": 27.53633962602873,
                        "fontWeight": "bold",
                        "paddingHorizontal": 7.5,
                        "textAlign": "center",
                      }
                    }
                  >
                    Return to
                    

                     login screen
                  </Text>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff59b",
                      "borderColor": "#104e01",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "space-evenly",
                      "maxWidth": "50%",
                      "minHeight": 80.03999999999999,
                      "minWidth": 285,
                      "opacity": 1,
                    }
                  }
                >
                  <Icon
                    name="login"
                    size={25}
                    style={
                      Object {
                        "color": "#104e01",
                      }
                    }
                  />
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Object {
                        "color": "#104e01",
                        "fontSize": 27.53633962602873,
                        "fontWeight": "bold",
                        "paddingHorizontal": 7.5,
                        "textAlign": "center",
                      }
                    }
                  >
                    Please
accept T&C
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
